File: road-rail-crossing-ng/src/app/services/road-rail-crossing-service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {RoadRailCrossing} from "../model/road-rail-crossing";
import {environment} from "../../environments/environment";
import {Observable} from "rxjs";

const URL = environment.road_rail_crossing;

@Injectable({
    providedIn: 'root'
})

export class RoadRailCrossingService {


    constructor(private http: HttpClient) {

    }

    public save(roadRailCrossing: RoadRailCrossing){
        return this.http.post(URL + '/save',roadRailCrossing);
    }

    public getOne(id: number): Observable<RoadRailCrossing> {
        console.log(id);
        return this.http.get<RoadRailCrossing>(URL + '/edit/' + id);
    }

    public delete(id: number){
        return this.http.delete(URL+'/delete/'+id);
    }

}

File: road-rail crossing/src/main/java/com/roadrail_crossing/risik/service/RoadRailCrossingService.java
package com.roadrail_crossing.risik.service;

import com.roadrail_crossing.risik.dto.RoadRailCrossingDto;
import com.roadrail_crossing.risik.entity.RoadRailCrossing;

public interface RoadRailCrossingService {
	void save(RoadRailCrossingDto dto);
	void delete(Long id);
	RoadRailCrossingDto getOne(Long id);
}
